python/tests/test_pyvw.py

```diff
@@ -182,4 +193,94 @@ def test_parse():
 
     ex = model.parse(["| a:1 b:0.5", "0:0.1:0.75 | a:0.5 b:1 c:2"])
     assert len(ex) == 2
+
+    # To test the prediction on multiline examples
+    finish = model.finish_example(ex)
+    assert model.predict(ex) == [0.0, 0.0]
     del model
+
+def test_numspace_id():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    nm1 = pyvw.namespace_id(ex, 2)
+    assert nm1.id == 2
+    assert nm1.ord_ns == 128
+    assert nm1.ns == '\x80' # Represents string of ord_ns
+    nm2 = pyvw.namespace_id(ex, 'my_example')
+    assert nm2.id is None # It is kept None as because we don't want to do linear search required to find it
+    assert nm2.ord_ns == 109
+    assert nm2.ns == 'm' # First character is considered as namespace id in case of string
+
+def test_example_namespace():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    nmid = pyvw.namespace_id(ex, 1)
+    exm = pyvw.example_namespace(ex, nmid, ns_hash=0)
```

Comment:
The hash of the namespace being 0 doesn't make sense?

Reply:
Can you please suggest for what value of `ns_hash` would it work?. Currently, I am unable to figure it out

Reply:
`ns_hash` should equal the output of the hash function from the string namespace. It's something that the parser would populate, that's why this particular object creation is hard to do. I'm not sure if this is something people would directly create?

Reply:
Okay, for test purpose (coverage) can we use this one then `exm = pyvw.example_namespace(ex, nmid, ns_hash=nmid.ord_ns)`. As I was a similar thing implemented in `example.feature()`